### PR TITLE
[CELEBORN-783] Revise the conditions for the `SortBasedPusher#insertRecord` method

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

[comment](https://github.com/apache/incubator-celeborn/commit/7adf1fca41cd34eb3e9ddf5cc8cd24674ab585b4?notification_referrer_id=NT_kwDOAIJHFbI3MDEyMzc0NzkwOjg1Mzc4Nzc#r121138008)

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

New UT

### CODE REVIEW

**Comment by cfmcgrady** [comment]
@waitinfuture @pan3793

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

File: client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java [hunk 1]
```diff
@@ -57,7 +57,6 @@ public class SortBasedPusher extends MemoryConsumer {
   private final int pushBufferMaxSize;
   private final long pushSortMemoryThreshold;
   final int uaoSize = UnsafeAlignedOffset.getUaoSize();
```
Comment (pan3793) [inline client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java:59]: static?

Comment (cfmcgrady) [reply client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java:59]: raised a follow-up https://github.com/apache/incubator-celeborn/pull/1704

**Pull request closed without merging**